pkg/pki/vaultpki/vault_pki.go

```diff
@@ -0,0 +1,287 @@
+			}
+			controllerutil.SetControllerReference(v.cluster, serverCert, scheme)
+			toCreate = append(toCreate, serverCert)
+		} else {
+			return errorfactory.New(errorfactory.APIFailure{}, err, "failed to check existence of server secret")
+		}
+	}
+
+	// the cruise control keystore volume
+	if err = v.client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      fmt.Sprintf(pkicommon.BrokerControllerTemplate, v.cluster.Name),
+			Namespace: v.cluster.Namespace,
+		},
+		clientSecret,
+	); err != nil {
+		if apierrors.IsNotFound(err) {
+			clientCert := &corev1.Secret{
+				ObjectMeta: templates.ObjectMeta(fmt.Sprintf(pkicommon.BrokerControllerTemplate, v.cluster.Name), pkicommon.LabelsForKafkaPKI(v.cluster.Name), v.cluster),
+				Data: map[string][]byte{
+					corev1.TLSCertKey:       controllerCert.Certificate,
+					corev1.TLSPrivateKeyKey: controllerCert.Key,
+					v1alpha1.CoreCACertKey:  controllerCert.CA,
+					v1alpha1.TLSJKSKey:      controllerCert.JKS,
+					v1alpha1.PasswordKey:    controllerCert.Password,
+				},
+			}
+			controllerutil.SetControllerReference(v.cluster, clientCert, scheme)
+			toCreate = append(toCreate, clientCert)
+		} else {
+			return errorfactory.New(errorfactory.APIFailure{}, err, "failed to check existence of client secret")
+		}
+	}
+
+	for _, o := range toCreate {
+		if err = v.client.Create(ctx, o); err != nil {
+			return errorfactory.New(errorfactory.APIFailure{}, err, "failed to create bootstrap secret", "secret", o.Name)
+		}
+	}
+
+	return
+}
+
+func (v *vaultPKI) reconcileBrokerCert(ctx context.Context, vault *vaultapi.Client) (*pkicommon.UserCertificate, error) {
+	return v.reconcileStartupUser(ctx, vault, pkicommon.BrokerUserForCluster(v.cluster))
+}
+
+func (v *vaultPKI) reconcileControllerCert(ctx context.Context, vault *vaultapi.Client) (*pkicommon.UserCertificate, error) {
+	return v.reconcileStartupUser(ctx, vault, pkicommon.ControllerUserForCluster(v.cluster))
+}
+
+func (v *vaultPKI) reconcileStartupUser(ctx context.Context, vault *vaultapi.Client, user *v1alpha1.KafkaUser) (cert *pkicommon.UserCertificate, err error) {
+	existing := &v1alpha1.KafkaUser{}
+	if err := v.client.Get(ctx, types.NamespacedName{Name: user.Name, Namespace: user.Namespace}, existing); err != nil {
+		if apierrors.IsNotFound(err) {
+			if err = v.client.Create(ctx, user); err != nil {
+				return nil, err
+			}
+		}
+	}
+	userSecret, v2, err := getSecret(vault, fmt.Sprintf("secret/%s", user.Spec.SecretName))
+	if err != nil {
+		return nil, err
+	} else if userSecret == nil || userSecret.Data == nil {
+		return nil, errorfactory.New(errorfactory.ResourceNotReady{}, errors.New("not ready"), "certificate is not ready")
+	}
+	cert, err = userCertForData(v2, userSecret.Data)
+	return
+}
+
+func (v *vaultPKI) initVaultPKI(vault *vaultapi.Client) error {
+	caPath := v.getCAPath()
+	// intermediatePath := v.getIntermediatePath()
+	userPath := v.getUserStorePath()
+	var err error
+
+	// Setup a PKI mount for the ca
```

Comment:
As a general comment for this PR (which is great in overall!) that I would leave out all Vault configuration parts from it because of these things require `sudo` Vault (admin) capabilities, for example manipulating mounts. I'm pretty sure that most people don't want to let the kafka-operator do things like that, so I would leave these parts to an exercise to the user, with vault-operator for example.

Reply:
I'm happy to find out how to configure Vault with our vault-operator to make these PKI backends available.

Reply:
This is done in the latest commit - I may need to update user docs more to reflect how vault needs to be setup. Also curious if there is a better way to handle passing vault's CA cert to the operator then what I'm doing right now in the helm chart with a secret/envvar.